app/javascript/packs/feedPreviewCards.jsx

```diff
@@ -0,0 +1,109 @@
+import { h, render } from 'preact';
+import { UserMetadata } from '../profilePreviewCards/UserMetadata';
+import { initializeDropdown } from '@utilities/dropdownUtils';
+
+async function populateMissingMetadata(metadataPlaceholder) {
+  const { authorId } = metadataPlaceholder.dataset;
+  const response = await fetch(`/profile_preview_cards/${authorId}`, {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+    credentials: 'same-origin',
+  });
+  const authorMetadata = await response.json();
+
+  // A given author may have multiple cards in the feed - populate their metadata everywhere it is missing
+  const allPlaceholdersForThisAuthor = document.querySelectorAll(
+    `.author-preview-metadata-container[data-author-id="${authorId}"]`,
+  );
+
+  for (const placeholder of allPlaceholdersForThisAuthor) {
+    render(
+      <UserMetadata {...authorMetadata} />,
+      placeholder.parentElement,
+      placeholder,
+    );
+
+    placeholder.parentElement.parentElement.style.setProperty(
+      '--card-color',
+      authorMetadata.card_color,
+    );
+
+    placeholder.remove();
+  }
+}
+
+function checkForPreviewCardDetails(event) {
+  const { target } = event;
+
+  if (target.classList.contains('profile-preview-card__trigger')) {
+    const metadataPlaceholder = target.parentElement.getElementsByClassName(
+      'author-preview-metadata-container',
+    )[0];
```

Comment:
Refactor (blocking): Prefer Element.closest()

```suggestion
    const metadataPlaceholder = target.closest('author-preview-metadata-container');
```

Reply:
`closest` doesn't work in this context, because it will traverse up the DOM tree to find a match. Whereas, I need to find an element that lives further _down_ the DOM tree 🙁